fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/service/MoneyHelperInitializationService.java

```
@@ -0,0 +1,115 @@
+
+            // Get rounding mode from configuration with fallback to default
+            int roundingModeValue = getRoundingModeFromConfiguration();
+
+            // Initialize MoneyHelper for this tenant
+            MoneyHelper.initializeTenantRoundingMode(tenantIdentifier, roundingModeValue);
+
+            log.info("MoneyHelper initialized for tenant '{}' with rounding mode: {}", tenantIdentifier, roundingModeValue);
+
+        } catch (Exception e) {
+            log.error("Failed to initialize MoneyHelper for tenant '{}'", tenantIdentifier, e);
+            throw new RuntimeException("Failed to initialize MoneyHelper for tenant: " + tenantIdentifier, e);
+        } finally {
+            // Clear tenant context
+            ThreadLocalContextUtil.clearTenant();
+        }
+    }
+
+    /**
+     * Check if MoneyHelper is initialized for a tenant.
+     *
+     * @param tenantIdentifier
+     *            the tenant identifier
+     * @return true if initialized, false otherwise
+     */
+    public boolean isTenantInitialized(String tenantIdentifier) {
+        return MoneyHelper.isTenantInitialized(tenantIdentifier);
+    }
+
+    /**
+     * Get the rounding mode from configuration with fallback to default. This method safely handles cases where
+     * configuration might not be available.
+     *
+     * @return the rounding mode value
+     */
+    private int getRoundingModeFromConfiguration() {
+        try {
+            GlobalConfigurationProperty roundingModeProperty = globalConfigurationRepository
+                    .findOneByNameWithNotFoundDetection(GlobalConfigurationConstants.ROUNDING_MODE);
+
+            if (roundingModeProperty != null && roundingModeProperty.getValue() != null) {
+                return roundingModeProperty.getValue().intValue();
+            }
+        } catch (Exception e) {
+            log.warn("Failed to read rounding mode from configuration: {}", e.getMessage());
```

I am not a big fan of swallowing this error...